smolyay/grid.py

```diff
@@ -0,0 +1,458 @@
+        n <= k_1 + k_2 + ... + k_n <= n + \mu
+
+    For instance:
+    ..math::
+        if n = 2 and \mu = 1, then:
+            if \sum_{1}^{n} k_i = 2:
+                (k_1 = 1, k_2 = 1)
+            if \sum_{i=1}^{n} k_i = 3:
+                (k_1 = 1, k_2 = 2)
+                (k_1 = 2, k_2 = 1)
+
+    The  :meth:`indices` generates the mentioned compositions,
+    and can expand or drop indices if necessary.
+    :meth:`__call__`  generates the indexes of
+    grid points (integer grids) depending on dimensionality and
+    exactness. First, as above, all possible combinations of
+    indices are computed, which then can be replaced with indexes based on the
+    levels. Then, grid points are generated by making all combinations of
+    indexes for each combination of levels' indices.
+
+    Once integer grid points are generated, actual grid points
+    (``grid_points``) and their corresponding basis functions
+    (``grid_points_basis``) can be made by simply replacing
+    the indexes with their corresponding points and basis functions.
+
+    :meth:`_update` updates the integer grid points, grid points and
+    their corresponding functions.
+
+    Parameters
+    ----------
+    nested_basis_set: :class:`NestedBasisSet`
+        points, basis functions and levels.
+
+    Example
+    -------
+    For instance, consider dimension (or n) = 2, and
+    exactness (or \mu) = 1 and Chebyshev's polynomial
+    of first kind, and its extremums as the basis function and points, then:
+    ..math::
+        levels = [[0], [1,2]]
+        level compositions = [([0],[0]), ([0],[1,2]), ([1,2],[0])]
+        then:
+        grid points indexes = [(0,0), (0,1), (0,2), (1,0), (2,0)]
+        points = [0, -1, 1]
+        basis = [:class:`ChebyshevFunction`, :class:`ChebyshevFunction`,
+                 :class:`ChebyshevFunction`]
+
+    Then:
+    ..math:
+        grid_points = [(0, 0), (0, -1), (0, 1), (-1, 0), (1, 0)]
+        grid points' functions = [(:class:`ChebyshevFunction`,
+        :class:`ChebyshevFunction`)
+        , (:class:`ChebyshevFunction`, :class:`ChebyshevFunction`),
+        (:class:`ChebyshevFunction`, :class:`ChebyshevFunction`),
+        (:class:`ChebyshevFunction`, :class:`ChebyshevFunction`),
+        (:class:`ChebyshevFunction`, :class:`ChebyshevFunction`)]
+
+    """
+
+    def __init__(self, nested_basis_set):
+        self._nested_basis_set = nested_basis_set
+        self._points = self._nested_basis_set.points
+        self._basis = self._nested_basis_set.basis_functions
+        self._levels = self._nested_basis_set.levels
+
+    @property
+    def nested_basis_set(self):
+        """:class:`NestedBasisSet`: Nested basis set (data structure)."""
+        return self._nested_basis_set
+
+    @nested_basis_set.setter
+    def nested_basis_set(self, nested_basis_set):
+        self._nested_basis_set = nested_basis_set
+
+    def indices(self, dimension, expand_indicy=[], drop_indicy=[]):
```

Comment:
We talked about removing the expand/drop here. Can you make that simplification before review?

Reply:
The point of this method was first storing the levels (which is also in data structure), and expanding or dropping levels. Since, we are simplifying it for now, I don't see any reason to keep this.